bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/console/ChannelConsoleCommandExtension.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.internal.console;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.events.ThingEventFactory;
+import org.eclipse.smarthome.io.console.Console;
+import org.eclipse.smarthome.io.console.extensions.AbstractConsoleCommandExtension;
+
+/**
+ * Console command extension to access channels
+ *
+ * @author Stefan Triller - Initial contribution
+ *
+ */
+public class ChannelConsoleCommandExtension extends AbstractConsoleCommandExtension {
```

Comment:
>  the class handles a command channel(s) that itself is able to support multiple sub-commands

@maggu2810 Would you see any other relevant sub-commands? Sending state updates and commands is something that should be done via items only (so that all "intermediate" logic within the framework is always respected). The other stuff you do with channels is to link them, to add or remove them from a thing, etc., but all of that is covered already in other commands.